pkg/controller/core/clusterqueue_controller.go

```diff
@@ -241,13 +243,47 @@ func (h *cqWorkloadHandler) requestForWorkloadClusterQueue(w *kueue.Workload) *r
 	}
 }
 
+// cqNamespaceHandler handles namespace update events.
+type cqNamespaceHandler struct {
+	qManager *queue.Manager
+	cache    *cache.Cache
+}
+
+func (h *cqNamespaceHandler) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (h *cqNamespaceHandler) Update(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	oldNs := e.ObjectOld.(*corev1.Namespace)
+	oldMatchingCqs := h.cache.MatchingClusterQueues(oldNs.Labels)
+	newNs := e.ObjectNew.(*corev1.Namespace)
+	newMatchingCqs := h.cache.MatchingClusterQueues(newNs.Labels)
+	cqs := sets.NewString()
+	for cq := range newMatchingCqs {
+		if !oldMatchingCqs.Has(cq) {
+			cqs.Insert(cq)
+		}
+	}
+	h.qManager.QueueInadmissibleWorkloads(cqs)
```

Comment:
Shall we consider just moving the workloads in the updated NS？But not for this pr.